src/docs/MLOS/MLOSUserGuide.adoc

```diff
@@ -166,6 +166,426 @@ E. Click the *Tasks* tab, select an *Preview_Results* task and click on the *Pre
+is by providing the name of the corresponding task, and the fork environment implementation after that:
+
+```text
+#%fork_env(name=TASK_NAME)
+containerName = 'activeeon/dlm3'
+dockerRunCommand =  'docker run '
+dockerParameters = '--rm '
+paHomeHost = variables.get("PA_SCHEDULER_HOME")
+paHomeContainer = variables.get("PA_SCHEDULER_HOME")
+proActiveHomeVolume = '-v '+paHomeHost +':'+paHomeContainer+' '
+workspaceHost = localspace
+workspaceContainer = localspace
+workspaceVolume = '-v '+localspace +':'+localspace+' '
+containerWorkingDirectory = '-w '+workspaceContainer+' '
+preJavaHomeCmd = dockerRunCommand + dockerParameters + proActiveHomeVolume + workspaceVolume + containerWorkingDirectory + containerName
+```
+
+A second way is by providing the name of the task, and the path of a _.py_ file containing the fork environment code:
+
+```python
+#%fork_env(name=TASK_NAME, path=PATH_TO/FORK_ENV_FILE.py)
+```
+
+==== Adding a selection script
+
+To add a selection script to a task, use the `#%selection_script()` pragma. A first way to do it is by
+providing the name of the corresponding task, and the selection code implementation after that:
+
+```python
+#%selection_script(name=TASK_NAME)
+selected = True
+```
+
+A second way is by providing the name of the task, and the path of a .py file containing the selection code:
+
+```python
+#%selection_script(name=TASK_NAME, path=PATH_TO/SELECTION_CODE_FILE.py)
+```
+
+==== Adding job fork environment and/or selection script
+
+If the selection scripts and/or the fork environments are the same for all job tasks, we can add them just once using
+the `job_selection_script` and/or the `job_fork_env` pragmas.
+
+Usage:
+
+For a job selection script, please use:
+
+```python
+#%job_selection_script([language=SCRIPT_LANGUAGE], [path=./SELECTION_CODE_FILE.py], [force=on/off])
+```
+
+For a job fork environment, use:
+
+```python
+#%job_fork_env([language=SCRIPT_LANGUAGE], [path=./FORK_ENV_FILE.py], [force=on/off])
+```
+
+The `force` parameter says if the pragma has to overwrite the task selection scripts or fork environment already set or not.
+
+==== Adding pre and/or post scripts
+
+Sometimes, specified scripts has to be executed before and/or after a particular task. To do that, the solution provides
+ `pre_script` and `post_script` pragmas.
+
+To add a pre-script to a task, please use:
+
+```python
+#%pre_script(name=TASK_NAME, language=SCRIPT_LANGUAGE, [path=./PRE_SCRIPT_FILE.py])
+```
+
+To add a post-script to a task, use:
+
+```python
+#%post_script(name=TASK_NAME, language=SCRIPT_LANGUAGE, [path=./POST_SCRIPT_FILE.py])
+```
+
+==== Create a job
+
+To create a job, use the `#%job()` pragma:
+
+```python
+#%job(name=JOB_NAME)
+```
+
+If the job has already been created, the call of this pragma would just rename the job already created by the new provided name.
+
+**Notice that it is not necessary to create and name explicitly the job. If not done by the user, this step is implicitly
```

Comment:
u meant assign a name to the job ?

Reply:
Nope. Creating the job is done by using the job pragma. It makes ProActive create a job related to all created tasks.

Reply:
assign a name  ok